internals/overlord/logstate/syslog/syslog.go

```diff
@@ -232,7 +235,8 @@ func (c *Client) buildSendBufferTCP() {
 		}
 
 		// TCP: Octet framing as per RFC 5425: <length> <message>
-		frameLength := len(priorityAndVersionPrefix) + len(entry.timestamp) + 1 + len(hostname) + 1 +
+		// RFC3339Nano time string ("2023-12-31T12:00:00.123456789Z") has length 30
+		frameLength := len(priorityAndVersionPrefix) + 30 + 1 + len(hostname) + 1 +
```

Comment:
Unfortunately, RFC3339Nano trims the trailing zeroes in the fractional seconds, so the fixed value of 30 isn't going to work. You can see this if you change one of the tests to 0 nanoseconds instead of 123456789, and the length is incorrect. What I suggest is that we do the `AppendTimestamp` logic in the caller, and pass `timeBuf` as an argument to `encodeOneEntry`. It's a little messy, but it'll work.

We should also change one of the tests to use 0 nanoseconds and fix the frame length, to test this case.